onadata/apps/logger/management/commands/update_attachment_storage_bytes.py

```diff
@@ -0,0 +1,177 @@
+            instance_ids_count = len(instance_ids)
+            max_ids_per_query = 5000
+            chunks = [
+                instance_ids[x:x + max_ids_per_query]
+                for x in range(0, instance_ids_count, max_ids_per_query)
+            ]
+            form_attachments_total = 0
+            for idx, chunk in enumerate(chunks):
+                if self.verbosity > 1:
+                    self.stdout.write(
+                        f'\tCalculating total: {idx+1}/{len(chunks)} instance ID'
+                        f' chunks'
+                    )
+                form_attachments = Attachment.objects.filter(
+                    instance_id__in=chunk
+                ).aggregate(total=Sum('media_file_size'))
+                if form_attachments['total']:
+                    form_attachments_total += form_attachments['total']
+
+            if form_attachments_total:
+                with transaction.atomic():
+                    if self.verbosity >= 1:
+                        self.stdout.write(
+                            f'\tUpdating xform attachment storage to '
+                            f"{form_attachments_total} bytes"
+                        )
+
+                    XForm.objects.select_for_update().filter(
+                        pk=xform['pk']
+                    ).update(
+                        attachment_storage_bytes=form_attachments_total
+                    )
+
+                total_per_user[xform['user_id']] += form_attachments_total
+            elif self.verbosity >= 1:
+                self.stdout.write('\tNo attachments found')
+
+            last_xform = xform
+
+        # need to call `update_user_profile()` one more time outside the loop
+        # because the last user profile will not be up-to-date otherwise
+        if last_xform:
+            self.update_user_profile(
+                last_xform, total_per_user[last_xform['user_id']]
+            )
+
+        if self.verbosity >= 1:
+            self.stdout.write('Done!')
+
+    def update_user_profile(self, xform: Dict, total: int):
+        user_id = xform['user_id']
+        username = xform['user__username']
+
+        if self.verbosity >= 1:
+            self.stdout.write(
+                f'Updating attachment storage total ({total} bytes) on '
+                f'{username}’s profile'
+            )
+
+        # Update user's profile (and lock the related row)
+        with transaction.atomic():
+            updates = {
+                'submissions_suspended': False,
+                'attachments_counting_status': 'complete',
+            }
+            UserProfile.objects.select_for_update().filter(
+                user_id=user_id
+            ).update(
+                attachment_storage_bytes=total,
+                metadata=ReplaceValues(
+                    'metadata',
+                    updates=updates,
+                ),
+            )
```

Comment:
This transaction makes sense to me :+1: but I believe `select_for_update()` still does not contribute anything.

Reply:
Oops :see_no_evil: I don't remember why it made sense. I've left a comment in my new review

Reply:
gone